Re-add IDBDatabase abort event

caugner opened the pull request:
Based on https://github.com/mdn/browser-compat-data/pull/15006.

<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->

#### Summary

Adds the `abort` event on `IDBDatabase`.

#### Test results and supporting details

The data was originally removed in https://github.com/mdn/browser-compat-data/pull/15006.

#### Related issues

Fixes: https://github.com/mdn/browser-compat-data/issues/15345

queengooborg commented on api/IDBDatabase.json:
```diff
@@ -98,6 +98,54 @@
+              "version_added": "22"
+            },
+            "ie": {
+              "version_added": "10",
+              "partial_implementation": true,
+              "notes": "Unknown limitations"
```
Probably would make more sense just to get rid of `partial_implementation` if we don't know what the limitations are.

```suggestion
              "version_added": "10"
```

ddbeck replied:
Since IE (intentionally) diverged from the spec (and never implemented the 2.0 spec), I think it's very likely that this event _does_ have weird or incomplete behavior (see https://github.com/Fyrd/caniuse/pull/532).

Instead of sweeping the data quality issues here under the rug, I'd suggest making it explicit:

```suggestion
              "version_added": "10",
              "partial_implementation": true,
              "notes": "Internet Explorer's IndexedDB implementation is known to be incomplete. The specific behavior of this event is poorly understood, but historic data suggests it has limitations. If you have more information about non-standard behaviors for this event, [file an issue or PR](https://github.com/mdn/browser-compat-data)."
```

caugner replied:
I would suggest to just leave it as is. Between removing the note altogether (including the partial implementation) and the verbose note, the current note is a good enough compromise.

ddbeck replied:
The status quo is fine with me.

ddbeck reviewed (commented):
Since this is the PR that's going to appear in the commit history, can you add some text to the PR description summarizing that we're doing something exceptional here, why, and that we're not planning to extend this to other PRs? I'd like some kind of paper trail that doesn't require reading the entirety of at least one issue (and probably more).

caugner commented:
> can you add some text to the PR description summarizing that we're doing something exceptional here, why, and that we're not planning to extend this to other PRs?

@ddbeck Can you elaborate the "something exceptional" bit? I have added a Motivation section in the PR now:

> The IDBDatabase event is defined in the spec, and implemented in Chrome, Firefox, and Safari, so the data shouldn't have been removed in the first place.
